Content.Server/Store/Components/StoreComponent.cs

```diff
@@ -59,6 +60,22 @@ public sealed partial class StoreComponent : Component
     [ViewVariables]
     public HashSet<ListingData> LastAvailableListings = new();
 
+    /// <summary>
+    ///     All current entities bought from this shop. Useful for keeping track of refunds and upgrades.
+    /// </summary>
+    [ViewVariables]
+    public List<EntityUid> BoughtEntities = new();
+
+    /// <summary>
+    ///     The total balance spent in this store. Used for refunds.
+    /// </summary>
+    public Dictionary<string, FixedPoint2> BalanceSpent = new();
+
+    [ViewVariables]
+    public bool RefundAllowed;
+
+    [ViewVariables]
+    public MapId StartingMap;
```

Comment:
I think a lot of these should be datafields? Have you tested that this works if you save and load?

Also please document `StartingMap` since it's not evident by the name what it is and why this comp needs to store it.

Reply:
How do you save and load?

Reply:
- `mapping` to make new map (saving the dev station throws an exception)
- `mapinit 2` to initialize the new map.
- Place the entities you want to test on the map.
- `savemap 2 /test.yml true` (bool forces the map to save even when it's initialized)
- `loadmap 3 /test.yml`
- While aghosted, open admin menu, objects tab, select maps, then right click the loaded map and follow.
- Test that everything works.

I did this with a syndicate uplink on master, and it seems to be broken.

![image](https://github.com/space-wizards/space-station-14/assets/48867431/83f18d45-39d0-4406-bfb0-5cae06ac5806)

Reply:
I've added datafields to these, other than starting map since it doesn't support it. The bug on master should be fixed at some point.